wordpress-mobile/WordPress-Android
# Issue/10726 site pages author picker

Fixes #10726 

This is a master-feature PR. The code has already been reviewed. However, it'd be great if someone with more experience with this feature take a look.

Goal of this series of PRs was to add an author filter to the pages view so that it matches functionality with the posts feature. The author name will be shown on the pages list when the author selector is set to Everyone. Search results are not filtered; however author name is shown.

There is one specific item that I would appreciate feedback on:
    - The Posts feature saves the author filter state in app prefs. I implemented Pages to use the same appPrefs property, so the filter selection is maintained across pages to posts. I was 50-50 on setting up a new property and I am open to suggestions.
    

**To test:**
- Launch WPAndroid
- Select a site not owned by you, but one in which you are an author
- Navigate to Pages
- Toggle the author selector to Me
- Ensure the author name is not shown
- Toggle the author selector to Everyone
- Ensure the author name is shown
- Tap the search icon
- Search for a page
- Ensure the author name is shown in the search results

PR submission checklist:
